operator/src/main/resources/scripts/introspectDomain.py

```diff
@@ -1056,6 +1059,221 @@ def writeListenAddress(self, originalValue, newValue):
       repVerb="\"add\""
     self.writeln("<d:listen-address f:combine-mode=" + repVerb + ">" + newValue + "</d:listen-address>")
 
+  def writeName(self, name):
+    self.writeln("<d:name>" + name + "</d:name>")
+
+  def writeProtocol(self, protocol):
+    self.writeln("<d:protocol f:combine-mode=\"add\">" + protocol + "</d:protocol>")
+
+  def writeListenPort(self, listen_port):
+    # offline WLST initializes int values to 0
+    if listen_port > 0:
+      self.writeln("<d:listen-port f:combine-mode=\"add\">" + str(listen_port) + "</d:listen-port>")
+
+  def writePublicAddress(self, public_address):
+    self.writeln("<d:public-address f:combine-mode=\"add\">" + public_address + "</d:public-address>")
```

Comment:
Will public_address be None and can sitconfig take None at this stanza (vs not set) ?

Reply:
MBean method for setting public address can handle null when looking at the generated MBean method implementation but added a conditional check.  Please review.